go/tools/gazelle/merger/merger_test.go

```diff
@@ -93,7 +93,37 @@ func TestMergeWithExisting(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	afterF, err := MergeWithExisting(newF)
+	afterF, err := MergeWithExisting(newF, tmp.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if s := string(bzl.Format(afterF)); s != expected {
+		t.Errorf("bzl.Format, want %s; got %s", expected, s)
+	}
+}
+
+func TestMergeWithExistingDifferentName(t *testing.T) {
+	tmp, err := ioutil.TempFile(os.Getenv("TEST_TMPDIR"), "BUILD")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmp.Name())
+	if _, err := io.WriteString(tmp, oldData); err != nil {
+		t.Fatal(err)
+	}
+	if err := tmp.Close(); err != nil {
+		t.Fatal(err)
+	}
+	newTmp, err := ioutil.TempFile(os.Getenv("TEST_TMPDIR"), "BUILD.bazel")
+	if err != nil {
+		t.Fatal(err)
```

Comment:
Looks like then only the last one should be t.Error -- if either this one or the next one fail, then the LHS might be nil, and continuing is suspect.